### PR TITLE
Fix/align all metadata

### PR DESCRIPTION
**Pull request by JosephKelly**
@StevenHee @john12321, this seemed as good a place as any to confirm the metadata that should be saved on a submitted/unsubmitted form.  Please approve if it's right.  If not, let me know what isn't as expected or, even better, push over the top.

It's probably [FormRPartAServiceImpl.java](https://github.com/Health-Education-England/tis-trainee-forms/pull/53/files#diff-bc8c011666df416bb83c4b18335123d9678c0aef356ad5795f2b2d40bdf67597) and [FormRPartBServiceImpl.java](https://github.com/Health-Education-England/tis-trainee-forms/pull/53/files#diff-913de133c3aed4bca0c2a6c56d59ce33ec2294346ef0ab5db98922fb01598db7) that you're most interested in.

### CODE REVIEW

File: src/test/java/uk/nhs/hee/tis/trainee/forms/service/FormRPartAServiceImplTest.java [hunk 1]
```diff
@@ -133,7 +145,19 @@ void shouldSaveSubmittedFormRPartA() {
     assertThat("Unexpected trainee ID.", savedDto.getTraineeTisId(), is(DEFAULT_TRAINEE_TIS_ID));
     assertThat("Unexpected forename.", savedDto.getForename(), is(DEFAULT_FORENAME));
     assertThat("Unexpected surname.", savedDto.getSurname(), is(DEFAULT_SURNAME));
-    verify(s3Mock).putObject(any(PutObjectRequest.class));
+    verify(s3Mock).putObject(putRequestCaptor.capture());
```
Comment (paulhoang-hee) [inline src/test/java/uk/nhs/hee/tis/trainee/forms/service/FormRPartAServiceImplTest.java:148]: nice

File: src/main/java/uk/nhs/hee/tis/trainee/forms/service/exception/ApplicationException.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.trainee.forms.service.exception;
+
+public class ApplicationException extends RuntimeException {
```
[the remark this answers is not in the document]
Comment (JosephKelly) [reply src/main/java/uk/nhs/hee/tis/trainee/forms/service/exception/ApplicationException.java:23]: Short answer: It satisfies a 'code smell'.
Longer answer: If the class is still there in a sprint or so; we should associate some service (Controller) handling with it to gracefully translate it for clients.

File: src/test/java/uk/nhs/hee/tis/trainee/forms/service/FormRPartAServiceImplTest.java (hunk 1 above)
Comment (paulhoang-hee) [reply src/test/java/uk/nhs/hee/tis/trainee/forms/service/FormRPartAServiceImplTest.java:148]: not entirely sure but the assertions around the `Unexpected Bucket Name.` bits sound like its part of an unhappy path test but the method name here `shouldSaveSubmittedFormRPartA` sounds like happy path

Comment (JosephKelly) [reply src/test/java/uk/nhs/hee/tis/trainee/forms/service/FormRPartAServiceImplTest.java:148]: I stuck with the convention `Unexpected {FIELD_NAME}.` for the display message if the assertion fails.  I did have to read it a second time.

**Review by StevenHee — APPROVED** [review]

**Pull request merged**